datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/cliUtils/DataframeOptions.scala

```diff
@@ -30,20 +32,51 @@ case class DataframeOptions(format: String, options: Map[String, String], path:
     if (format == "jdbc") dfReader.load()
     else dfReader.load(path)
 
-  private def save[T](dfWriter: DataFrameWriter[T], extraPath: String = ""): Unit =
+  private def save[T](dfWriter: DataFrameWriter[T], endPath: String = ""): Unit =
     if (format == "jdbc") dfWriter.save()
-    else dfWriter.save(s"$path$extraPath")
+    else dfWriter.save(endPath)
+
+  def getUniqueFilePath(extraPath: String, conf: Configuration): String = {
+    val fs = FileSystem.get(conf)
+    val basePath = s"$path$extraPath"
+
+    @scala.annotation.tailrec
+    def appendNumberAndTest(namePt1: String,
```

Comment:
What about using UUID? Or timestamp of great precision (yyyyMMddHHmmssSSS)?
instead of trying always from one. I am aware that on modern systems it shouldn't be ever too long, but in principle. What if the destination is far away with long ping?

Reply:
Did we finish this in the offline discussions?